### PR TITLE
[Docs] Fix hw, comb and sv dialect links

### PR DESCRIPTION
**Pull request by bluewww**
Seems like these links were not updated when the corresponding files were moved into separate subfolders.

### CODE REVIEW

**Comment by mortbopet** [comment]
Thanks! 

Would you like to merge this yourself? If so, you'll need to request commit access (below). If not, I can merge it for you. Requesting access is a good idea if you plan on contributing further to CIRCT! 👍 

https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access

**Comment by bluewww** [comment]
Thanks! In that case I will ask for commit access.

**Review by mortbopet — APPROVED** [review]

**Pull request merged**
